src/Compilers/Core/MSBuildTask/ValidateBootstrap.cs

```diff
@@ -87,17 +87,18 @@ public override bool Execute()
             return allGood;
         }
 
-        private static string NormalizePath(string path)
+        private static string? NormalizePath(string? path)
         {
+            // PROTOTYPE(NullableDogfood): string.IsNullOrEmpty needs annotation
             if (string.IsNullOrEmpty(path))
             {
                 return path;
             }
 
-            var c = path[path.Length - 1];
+            var c = path![path!.Length - 1];
```

Comment:
💭 Isn't the following sufficient?

```
path[path!.Length - 1]
```

Reply:
Current semantics of `!` is to suppress conversion and assignment warnings, but it does not affect the state of the expression it is applied to.